Alarm codec

dpino opened the pull request:
Same as #877 but target remote branch is lwaftr-alarms.

dpino committed:
Implement alarm_codec

dpino committed:
Basic alarms channel mechanism

dpino committed:
Decompress paramaters individually in raise-alarm and clear-alarm encoding/decoding

dpino committed:
Group decompressed arguments into key and args

dpino committed:
Notify raise and clear alarms in ARP app

dpino committed:
Create alarms-channel if it doesn't exist

dpino committed:
Emit alarm notification in ARP app only if set

dpino committed:
Encode raise-alarm and clear-alarm directly

wingo commented on src/apps/ipv4/arp.lua:
```diff
@@ -141,14 +144,23 @@ function ARP:new(conf)
       o.arp_request_pkt = make_arp_request(o.self_mac, o.self_ip, o.next_ip)
       self.arp_request_interval = 3 -- Send a new arp_request every three seconds.
    end
+   o.alarm_notification = conf.alarm_notification
```
This will happen automatically because of line 138.

wingo commented on src/program/lwaftr/run/run.lua:
```diff
@@ -158,6 +158,7 @@ function run(args)
    end
 
    if opts.reconfigurable then
+      conf.alarm_notification = true
```
Because the conf is a product of a yang schema I hesitate to do this.  Instead I would have setup.lua arrange to ensure that ARP is created with alarm notification enabled.

dpino committed:
Make alarm_notification flag a local

Flag should be passed to all apps that can raise alarms.  Flag is always
true when running lwAFTR in reconfigurable mode.  Otherwise false.

wingo reviewed (approved):
lgtm, thanks!
